lib/exception_handling.rb

```diff
@@ -324,24 +324,28 @@ def ensure_completely_safe(exception_context = "", log_context = {})
     end
 
     def escalate_to_production_support(exception_or_string, email_subject)
+      ActiveSupport::Deprecation.warn('escalate_to_production_support is deprecated as of exception_handling 3.0')
       production_support_recipients or raise ArgumentError, "In order to escalate to production support, you must set #{name}.production_recipients"
       ex = make_exception(exception_or_string)
       escalate(email_subject, ex, last_exception_timestamp, production_support_recipients)
     end
 
     def escalate_error(exception_or_string, email_subject, custom_recipients = nil, log_context = {})
+      ActiveSupport::Deprecation.warn('escalate_error is deprecated as of exception_handling 3.0')
       ex = make_exception(exception_or_string)
       log_error(ex, **log_context)
       escalate(email_subject, ex, last_exception_timestamp, custom_recipients)
     end
 
     def escalate_warning(message, email_subject, custom_recipients = nil, log_context = {})
+      ActiveSupport::Deprecation.warn('escalate_warning is deprecated as of exception_handling 3.0')
       ex = Warning.new(message)
       log_error(ex, **log_context)
       escalate(email_subject, ex, last_exception_timestamp, custom_recipients)
     end
 
     def ensure_escalation(email_subject, custom_recipients = nil, log_context = {})
+      ActiveSupport::Deprecation.warn('ensure_escalation is deprecated as of exception_handling 3.0')
```

Comment:
All of these deprecations can be dried up a little bit to help with more uniform messaging. If you put the following after all the methods are defined, it will work the same as what you have here

```
deprecate :escalate_to_production_support,
          :escalate_error,
          :escalate_warning,
          :ensure_escalation,
          deprecator: ActiveSupport::Deprecation.new('3.0', 'ExceptionHandling')
```

Reply:
Ahhh I figured there'd be a cleaner way to do this. Thanks!